app/questionnaire/path_finder.py

```diff
@@ -22,9 +22,9 @@ def __init__(
         self,
```

Comment:
I think we should also move `is_path_complete` and `get_first_incomplete_location` into Router. `get_first_incomplete_location` is called from Router anyways and `is_path_complete` is called from the block handlers that have access to a Router.

Reply:
These now live in `Router`.